staging/src/k8s.io/client-go/tools/clientcmd/api/types.go

```diff
@@ -172,6 +193,29 @@ type ExecConfig struct {
 	APIVersion string `json:"apiVersion,omitempty"`
 }
 
+var _ fmt.Stringer = new(ExecConfig)
+var _ fmt.GoStringer = new(ExecConfig)
+
+// GoString implements fmt.GoStringer and sanitizes sensitive fields of
+// ExecConfig to prevent accidental leaking via logs.
+func (c ExecConfig) GoString() string {
+	return c.String()
+}
+
+// String implements fmt.Stringer and sanitizes sensitive fields of ExecConfig
+// to prevent accidental leaking via logs.
+func (c ExecConfig) String() string {
+	args := "<nil>"
+	if len(c.Args) > 0 {
+		args = "--- REDACTED ---"
+	}
+	env := "<nil>"
+	if len(c.Env) > 0 {
+		env = "--- REDACTED ---"
+	}
+	return fmt.Sprintf("{Command: %q, Args: %s, Env: %s, APIVersion: %q}", c.Command, args, env, c.APIVersion)
```

Comment:
is it ok for GoString() to not return valid go syntax? can we redact args and env in syntactically valid ways?

Reply:
The interface docs don't require valid Go syntax. It's the default behavior but nothing should depend on it.
But I changed the format to look as close to valid syntax as i could anyway.
WDYT?